### ISSUE TITLE
Usability: Must home before starting print

### ISSUE THREAD

**Issue by nstephenh** [KevinOConnor/klipper#2740]
When starting a print with octoklipper, if I haven't homed the printer since the last print completed, the printer returns a Must home axis first: 0.000 180.000 20.000 [61085.740] command and disconnects. When reconnecting it returns a m112 error and disconnects. I've had this happen multiple times in a row. Eventually it will reconnect and restarting the firmware and re-homing before starting the print works fine. 

My thoughts, after a day or so with klipper, is that this doesn't seem like an optimized workflow. My marlin printers auto-home as soon as I click print, instead of emergency stopping the printer and preventing it from re-connecting.

**Comment by klipper-gitissuebot** [comment]
Hi @nstephenh,

It did not look like there was a Klipper log file attached to this ticket. The log file has been engineered to answer common questions the Klipper developers have about the software and its environment (software version, hardware type, configuration, event timing, and hundreds of other questions).

Unfortunately, too many people have opened tickets without providing the log. That consumes developer time; time that would be better spent enhancing the software. If this ticket references an event that has occurred while running the software then the Klipper log must be attached to this ticket. Otherwise, this ticket will be automatically closed in a few days.

For information on obtaining the Klipper log file see: https://github.com/KevinOConnor/klipper/blob/master/docs/Contact.md

The log can still be attached to this ticket - just add a comment and attach the log to that comment.

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being.

**Comment by GarthSnyder** [comment]
This isn't normal. You are sending a G-code command that requires the axes to be homed, probably from your slicer or OctoPrint preamble. Look in the OctoPrint's Terminal pane - what is the actual command that precedes the "must home" error message?

Klipper seems to be a bit more stringent about requiring homing than Marlin, which explains why the same code might work under Marlin but not Klipper. My guess is that when you find and understand the issue, you will wonder why Marlin allows it without homing.

**Comment by BlackStump** [comment]
In Octoprint the default settings do not give a new user a great experience, so you need to go through Octoprints settings to make sure it does not issue random M112 also as per the klipper install instructions make sure you did not miss this step in configuring Octoprint for Klipper
>In the Settings tab, navigate to the “Behavior” sub-tab and select the “Cancel any ongoing prints >but stay connected to the printer” option. Click “Save”.

You do not want Octoprint disconnecting from Klipper, Klipper has all the safety functions you need so avoid Octoprint taking on that role.

**Comment by nstephenh** [comment]
@BlackStump I disabled that setting and it now doesn't disconnect. I still have to home it manually though. Thank you.

**Comment by Nandox7** [comment]
> This isn't normal. You are sending a G-code command that requires the axes to be homed, probably from your slicer or OctoPrint preamble. Look in the OctoPrint's Terminal pane - what is the actual command that precedes the "must home" error message?
> 
> Klipper seems to be a bit more stringent about requiring homing than Marlin, which explains why the same code might work under Marlin but not Klipper. My guess is that when you find and understand the issue, you will wonder why Marlin allows it without homing.

Note quite as I've used not Mrlin but RepetierFW for long time where this is possible and instead I wonder why Klipper does not allow it.
While is not something you'd be using for every single print but there are several instances where this is helpful. 
Like you need to emergency stop a print, carriage is somewhere in the bed, with RepetierFW I'd simply raise the Z, clean whatever mess was made, home and continue. With klipper I have the manually raise the Z axis (in my case rotating both leadscrews manually, clean the mess, hope the X axis is still leveled, home and level out the X axis.

**Comment by GarthSnyder** [comment]
> @Nandox7  wrote: ...there are several instances where this is helpful. Like you need to emergency stop a print, carriage is somewhere in the bed, with RepetierFW I'd simply raise the Z, clean whatever mess was made, home and continue. With klipper I have the manually raise the Z axis (in my case rotating both leadscrews manually, clean the mess, hope the X axis is still leveled, home and level out the X axis.

I believe you can use the MANUAL_STEPPER command to achieve this effect in Klipper.

**Comment by Nandox7** [comment]
Just read on MANUAL_STEPPER.
Doesn't seem to be an option, if I understood correct it would be for stepper drivers that are working outside of the kinematics definition of Klipper. In my example all the steppers are all already in use.

# Manual steppers (one may define any number of sections with a
# "manual_stepper" prefix). These are steppers that are controlled by
# the MANUAL_STEPPER g-code command. For example: "MANUAL_STEPPER
# STEPPER=my_stepper MOVE=10 SPEED=5". See the docs/G-Codes.md file
# for a description of the MANUAL_STEPPER command. The steppers are
# not connected to the normal printer kinematics.

**Comment by klipper-gitissuebot** [comment]
It appears the Klipper log file was not provided.
I'm closing this now, sorry. Feel free to attach the log to this ticket sometime in the future, or create a new issue once you can provide the log file.
